lib/ansible/module_utils/vmware.py

```diff
@@ -1000,6 +1000,36 @@ def get_vm_path(content, vm_name):
             folder_name = '/' + folder_name
         return folder_name
 
+    def get_vm_template(self, template_name=None):
+        """
+        Function to find the virtual machine template using name.
+        Args:
+            template_name: Name of template
+
+        Returns: virtual machine template object
+
+        """
+        template_obj = None
+
+        if template_name:
+            objects = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
+            templates = []
+
+            for temp_vm_object in objects:
+                if len(temp_vm_object.propSet) != 1:
+                    continue
+                for temp_vm_object_property in temp_vm_object.propSet:
```

Comment:
Trying to understand... If the length of `temp_vm_object.propSet` is not exactly 1, you loop over `temp_vm_object.propSet`? Why not just use `temp_vm_object.propSet[0]`?

Reply:
Yes. Let us say propSet does not content any property then this will be helpful. I encountered Index error with some VM in testing environment so preferred to keep this way.

Reply:
ok

Reply:
@pdellaert in python you can't access [set](https://docs.python.org/3.6/library/stdtypes.html#set-types-set-frozenset) elements by index:
```
>>> a=set((1, 3, 2))
>>> a
{1, 2, 3}
>>> a[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'set' object does not support indexing
```